### PR TITLE
[WIP] Get rid of generic needle text-login

### PR DESCRIPTION
**Pull request by SergioAtSUSE**
- Related ticket: https://progress.opensuse.org/issues/34471
- Needles: https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/360
- Verification run:
  - external_iso: http://10.160.66.74/tests/228#step/copy_iso_to_external_drive/28
  - minimalx: http://10.160.66.74/tests/229#step/consoletest_setup/33
  - zypper_up:
    - TW: http://10.160.66.74/tests/251#step/zypper_up/1
    - Leap 15: http://10.160.66.74/tests/252#step/zypper_up/1
    - Leap 42.3 textmode: http://10.160.66.74/tests/261#step/zypper_up/1
  - (More coming soon)

### CODE REVIEW

File: lib/susedistribution.pm [hunk 1]
```diff
@@ -503,7 +503,6 @@ sub activate_console {
             $nr = get_root_console_tty if ($name eq 'root');
             $nr = 5 if ($name eq 'log');
             my @tags = ("tty$nr-selected", "text-logged-in-$user");
-            push(@tags, 'text-login') unless $args{ensure_tty_selected};
```
Comment (okurz) [inline lib/susedistribution.pm]: If you remove this then you should remove the block in line R509-R525 as well.

Both https://w3.nue.suse.com/~okurz/openqa_opensuse_org_status.html as well as https://w3.nue.suse.com/~okurz/openqa_suse_de_status.html reference existing tests which hit this code branch so I guess you can start with these scenarios first, see why tty$nr-selected does not match there, make sure they do and then remove the text-login needle. As I saw in another comment by you that you identified test suites which use text-login in another place maybe it makes sense to rename the used tag either here or in the other places to separate the issues

File: tests/console/copy_iso_to_external_drive.pm [hunk 2]
```diff
@@ -43,7 +43,7 @@ sub run {
 sub post_run_hook {
     #prepare environment for next test
     type_string "logout\n";
-    assert_screen "text-login";
+    assert_screen "tty6-selected";
```
Comment (okurz) [inline tests/console/copy_iso_to_external_drive.pm]: please don't hard-code the tty$nr-selected in the test modules. This is what we have the `select_console`-calls for. So, let me think, why do we have this `assert_screen` here?

commit 6384c26c
Author: Joachim Rauch <jrauch@suse.com>
Date:   Fri Aug 18 12:41:57 2017 +0200

    Fix ISO_IN_EXTERNAL test
    
    Fixed the problem when rebooting to the installation.

introduced it. That was in PR https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3425 . Unfortunately the PR only states "Fixed the problem when rebooting to the installation" without stating in detail what actually was the problem. Honestly I think this line can simply be removed. It should not matter if the actual "logout" appears in the background while we already switch the x11 console

**Comment by okurz** [comment]
Please rebase and update your PR with a message about the current status. Could be @jknphy already did some related changes? Please crosscheck

**Comment by okurz** [comment]
Isn't this relevant still?

**Comment by SergioAtSUSE** [comment]
I think my changes were in the wrong direction

**Pull request closed without merging**